app/jobs/autoscaling.py

```diff
@@ -260,227 +61,313 @@ def get_instance_tags():
-    print "From cloud environments: {num_real_jobs}".format(num_real_jobs=num_real_jobs)
-    print "From development environments: {num_development_jobs}".format(num_development_jobs=num_development_jobs)
-    can_scale = permission_to_scale(gsnap_asg, my_environment)
-    if not can_scale:
-        print "Scaling by agents of {my_environment} is not permitted.".format(my_environment=my_environment)
-        # when debugging is enabled and we can't scale we print the scaling commands without executing them
-        # when debugging is disabled and we can't scale we exit early here
-        if not DEBUG:
-            return
-    tag_list = get_instance_tags()
-    gsnap_tags = find_tags_for_instances_in(gsnap_asg, tag_list)
-    rapsearch_tags = find_tags_for_instances_in(rapsearch2_asg, tag_list)
-    if num_real_jobs == 0 and num_development_jobs == 0:
-        set_desired_capacity(gsnap_asg, gsnap_instance_name, gsnap_tags, draining_tag, exactly(0), can_scale)
-        set_desired_capacity(rapsearch2_asg, rapsearch_instance_name, rapsearch_tags, draining_tag, exactly(0), can_scale)
-    elif num_real_jobs == 0 and num_development_jobs > 0:
-        count_running_servers = get_previous_desired(gsnap_asg) + get_previous_desired(rapsearch2_asg)
-        print "Only development environments are reporting in-progress jobs, and {crs} servers are running.".format(crs=count_running_servers)
-        if count_running_servers > 2 or random.random() < 0.2:
-            crbj = count_running_batch_jobs()
-            if crbj:
-                print "{crbj} jobs are running in aws batch queues".format(crbj=crbj)
-                # This is an unsafe scaling operation, but the only jobs that can get hurt
-                # are development jobs.  Just rerun those.
-                set_desired_capacity(gsnap_asg, gsnap_instance_name, gsnap_tags, draining_tag, exactly(1), can_scale)
-                set_desired_capacity(rapsearch2_asg, rapsearch_instance_name, rapsearch_tags, draining_tag, exactly(1), can_scale)
-            elif crbj == 0:
-                print "No jobs are running in batch."
-                # in this case num_development_jobs are either zombies or jobs caught in transition
-                # between pipeline stages;  the non-zombies won't be hurt, only delayed by this
-                set_desired_capacity(gsnap_asg, gsnap_instance_name, gsnap_tags, draining_tag, exactly(0), can_scale)
-                set_desired_capacity(rapsearch2_asg, rapsearch_instance_name, rapsearch_tags, draining_tag, exactly(0), can_scale)
+    # If you are running into problems with this autoscaler, just delete this tag from
+    # any affected autoscaling group, and it will be left alone.  The value is a list
+    # of environments permitted to trigger scaling, usually just "prod".
+    scaling_permission_tag = 'IDSeqEnvsThatCanScale'
```

Comment:
Oh wait this comment explains why I had originally built "can_scale" this way.  I had thought there could be bugs in this script and we could just take over manually in that case by removing the tag from the ASG.   You know what... that was a good idea... :)   Because you don't want to have to force a production web push to deal with pipeline operational problems.   I think perhaps this is worth leaving in, after all.

Reply:
Yes!